bot/cogs/reddit.py

```diff
@@ -290,4 +290,7 @@ def channel(self) -> TextChannel:
 
 def setup(bot: Bot) -> None:
     """Load the Reddit cog."""
+    if None in (RedditConfig.client_id, RedditConfig.secret):
```

Comment:
I missed this: if unconfigured, these will be empty strings, not `None`.

Reply:
It's `None` when excluded from the .env, empty strings if just the values are unset.
Empty string get caught when the cog tries to authenticate and is then unloaded.

Reply:
Under docker, they default to empty strings when excluded from `.env`. I think it would make sense to check for both in the setup.

Reply:
Sorry, I realize now that the issue this addresses is that the cog crashes if these two sneak in as `None`s. I assume that this is then a problem that only happens when running without docker.

I still think that it'd make sense to unload & log if they are empty strings, but I'll re-approve now. If you don't want to do that, then you can mark this as resolved.

Reply:
At a minimum it'll save 3 requests which are done before the cog is unloaded so won't hurt.